App.js

```diff
@@ -1,129 +1,148 @@
-  test: (node) => node.type === 'text' && node.parent !== null && node.data?.trim().length >= 0
-}
-
-const ParagraphBlock = ({ children }) => {
-  return <Text>[Text Component]{children}[Text Component]</Text>
+  test: (node) =>
+    node.type === "text" &&
+    node.parent !== null &&
+    node.data?.trim().length >= 0,
 };
 
-const ListWrapperBlock = ({ children }) => {
-  return <View><Text>[React Native List Component]</Text>{children}<Text>[React Native List Component]</Text></View>
-}
-
-const ListItemWrapperBlock = ({ children }) => {
-  return <Text>[React Native List Item]{children}[React Native List Item]</Text>
-}
-
-/**
- * In this example FlatListBlock goes over its children to build an actual list.
- * 
- * This assumes this the list is text only
- * 
- * @param {*} param0 
- * @returns 
- */
-const FlatListBlock = ({ children }) => {
-  // note: at this point children has already recursively gone through the "dom-to-react" conversion
-  const items = children.filter((node) => node.type === 'li').map(child => { return child.props.children; });
-  
-  const renderItem = ({ item }) => {
-    return (
-      <Text>{item}</Text>
-    );
-  }
-  return <FlatList data={items} renderItem={renderItem} />
-}
-
 /**
  * A noop block
- * 
+ *
  * You can also use this to create a "allowList" of blocks and ignore blocks you don't want to render
- * 
- * @returns 
+ *
+ * @returns
  */
-const EmptyBlock = () => <></>
+const EmptyBlock = () => <></>;
 
 EmptyBlock.defaultProps = {
   /**
    * Essentially catches any empty text nodes to make sure they don't get rendered under a <View> component
-   * 
-   * @param {*} node 
-   * @returns 
+   *
+   * @param {*} node
+   * @returns
    */
   test: (node) => {
-    return node.type === 'text' && (node.parent === null || node.data?.trim().length === 0);
+    const allowedBlocks = ["core/paragraph", "core/heading", "core/image"];
+
+    // if this isn't an allowed block catch it
+    // if this is an allowed block it should have been handled at this ppint
+    if (
+      node.type !== "text" &&
+      !allowedBlocks.some((blockName) => isBlockByName(node, blockName))
+    ) {
+      return true;
+    }
+
+    return (
+      node.type === "text" &&
+      (node.parent === null || node.data?.trim().length === 0)
+    );
+  },
+};
+
+const RNParagraphBlock = ({ children }) => {
+  return <Text>{children}</Text>;
+};
+
+const RNHeadingBlock = ({ children, level }) => {
+  return <Text style={styles[`headingLevel${level}`]}>{children}</Text>;
+};
+
+const RNImageBlock = ({ children, domNode, height, width, src }) => {
+  const caption =
+    domNode.children.find((el) => el.name === "figcaption")?.firstChild?.data ??
+    "";
+  return (
+    <>
+      <Image
+        style={[{ width: width, height: height }]}
+        source={{
+          uri: src,
+        }}
+      />
+      <Text>{caption}</Text>
+    </>
+  );
+};
+
+function SinglePostComponent() {
+  const { loading, data } = useFetchPost({
```

Comment:
`useFetchPost` is one of the custom hooks exposed by the package. It's build on top of vercel `useSWR` library.